### PR TITLE
initial version of msgFunction

### PR DESCRIPTION
**Pull request by Uzlopak**
@AbdelrahmanHafez 
@vkarpov15 

I wonder if you can figure out what I did here ;).

### CODE REVIEW

**Comment by Uzlopak** [comment]
Dont merge this yet ;)

**Comment by vkarpov15** [comment]
Faster handling of `{VALUE}` and other properties in error messages I presume?

**Comment by Uzlopak** [comment]
Exactly :). But using this everywhere relevant seems to be kind of hard...

**Comment by AbdelrahmanHafez** [comment]
Thanks @Uzlopak 

I can see that this is related to parsing validation error messages, but I'm finding it a bit difficult to understand how this is meant to be used.

Other than verifying that the code works, unit tests also show examples of how the author meant for the code to be used.
Would happily help replace the code all over the place to use this once I understand how to use it.

File: lib/error/msgFunction.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+'use strict';
+
+module.exports = function msgFunction(msg) {
```
Comment (AbdelrahmanHafez) [inline lib/error/msgFunction.js:3]: I've always found that it helps in code readability to name functions as verbs:
```js
const replaceMessage = getMessageReplacer(msg); // vs msgFunction(msg)
const parsedMesage = replaceMessage(someVariable);
```

**Review by AbdelrahmanHafez — CHANGES_REQUESTED** [review]

**Pull request closed without merging**